### ISSUE TITLE
BUG: It happens only on scipy not on any other package - autocomplete not possible in IDLE. ipython 

### ISSUE THREAD

**Issue by AnthonyTheKoala** [scipy/scipy#15944]
### Describe your issue.

I have this problem in regards "autocompletion" of scipy.

When one imports a package, any package, you'd expect to see after the fullstop, a list of functions.

### Attempts to uninstall and reinstall 
But no matter what one does:  
pip  uninstall scipy --upgrade
**Then either:**
pip install scipy --upgrade
pip install scipy --force-reinstall 
pip install --ignore-installed scipy
pip install --upgrade --no-deps --force-reinstall scipy
pip install --no-deps --ignore-installed scipy

**Results of uninstalling and reinstalling - regardless of reinstallation method:**
That is instead of the many functions of scipy, only functions I get are this:
![image](https://user-images.githubusercontent.com/26536316/161892459-58dc824e-52c6-4884-a759-16dff27c3429.png)

In other words I don't get some of the many functions such as scipy.e and scipy..pi. 
For example I only get this:

![161891760-db71eac3-3f3a-4951-8345-4e3110eed140](https://user-images.githubusercontent.com/26536316/161892186-b3d85ca1-64ef-4a2b-8bb0-12d4b0d4a22c.png)

### Other information:
scipy.__version__ = '1.8.0'
numpy.__version__ = '1.22.3'
Python version = 3.9.12 = sys.version_info = sys.version_info(major=3, minor=9, micro=12, releaselevel='final', serial=0)
OS: Windows: 19043.985

**Note** that this behaviour is only occurring with scipy, NOT any other package. That is autocompletion occurs successfully with other packages but not for scipy.

Thank you,
Anthony of Sydney

### Reproducing Code Example

```python
Python 3.9.12 (tags/v3.9.12:b28265d, Mar 23 2022, 23:52:46) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license()" for more information.
>>> import scipy
>>> # After you press the dot for scipy.  you expect to see many functions.
>>> # NOT two functions. 
>>> dir(scipy)
['LowLevelCallable', 'test']
>>> # Yet we know that scipy has more functionality.
>>> print(scipy.e)
2.718281828459045
>>> print(scipy.pi)
3.141592653589793
>>>
```

### Error message

```shell
There were no error messages.
```

### SciPy/NumPy/Python version information

sys.version_info(major=3, minor=9, micro=12, releaselevel='final', serial=0) = 3.9.12

**Comment by mdhaber** [comment]
Please import individual subpackages before trying autocomplete. For details, see[SciPy Organization](https://docs.scipy.org/doc/scipy/tutorial/general.html#scipy-organization).
![image](https://user-images.githubusercontent.com/6570539/161909433-f3b0d053-cab2-4461-b5de-83ca1b042faa.png)

**Comment by AnthonyTheKoala** [comment]
Dear sir,
Thank you for referring me to the document at https://docs.scipy.org/doc/scipy/tutorial/general.html#scipy-organization .

Perhaps I should have expressed the problem more precisely.

I should not have to do the from __scipy import linalg_, optimize_ in order explore the available functions of scipy NOR read the documentation as suggested in your last comment.

(1) Applying the last comment produce produced this result
![image](https://user-images.githubusercontent.com/26536316/161915337-3c725756-b315-494b-834c-f937c4315fe5.png)

Where did  that DLL issue came about?

(2) This problem is only occurring in scipy.

I was able to access functions of scipy by merely importing _import scipy_.

First I should not have have to use the _from scipy import linalg_

Other packages work with autocompletion, such as from numpy:

![image](https://user-images.githubusercontent.com/26536316/161914006-c5d167f9-c8bf-4932-8754-4c64935b32b0.png)

It is only with scipy that I have the problem.

But with scipy I only get 

![image](https://user-images.githubusercontent.com/26536316/161914134-d2cd3881-0304-4f35-9719-46a5da2b9bb1.png)

There is a bug in your scipy that is not in numpy.

Please help 
Thank you,
Anthony of Sydney

**Comment by AnthonyTheKoala** [comment]
Autocompletion works in Spyder but not in IDLE as demonnstrated before
From Spyder
![image](https://user-images.githubusercontent.com/26536316/161930199-5e36cb0e-ab2d-4cda-891e-ec8247b38c71.png)

Does anybody understand me? 

Why isn't it working in IDLE but works in spyder?

Thank you,
Anthony of Sydney

**Comment by mdhaber** [comment]
Please see gh-11774 for one of the other times this was discussed. There is some work towards [lazy importing](https://scientific-python.org/specs/spec-0001/) that would not require importing subpackages (see gh-15320), but for now, importing each subpackage is the recommended way of using SciPy. 

@stefanv do I understand that submodule names should show up in every IDE starting in SciPy 1.9.0? Do you know why the behavior might be be inconsistent until then?

**Comment by AnthonyTheKoala** [comment]
Dear sirs,
Thank you for your time.
I get autocompletion from MS Visual Studio Code,
![image](https://user-images.githubusercontent.com/26536316/161986680-4bba1f59-b0ec-4727-852f-66e110bee53d.png)

There is no autocompletion in MS VIsual Studio 2022's Python interactive terminal.

Hope that it is helpful for you,

Thank you,
Anthony of Sydney

**Comment by rkern** [comment]
IDLE's autocompletion is not as sophisticated as those others. It doesn't try to show unimported subpackages. Different IDEs are implemented differently. That has more to do with the IDEs than `scipy`.

In some ways, it's more correct than the other IDEs. You cannot just `import scipy` and use `scipy.linalg.whatever()`. `scipy.linalg` isn't available until you do `from scipy import linalg` or `import scipy.linalg`. That's how `scipy` is designed. `numpy` is designed differently.

Now, IDLE probably should be showing some more symbols that `scipy/__init__.py` imports from `numpy`, but it looks like you have a misbuilt `scipy`, which may also be causing issues with the autocompletion functionalities. That's the "DLL load failed" error message. We can't diagnose here, but you will have to revisit how you installed scipy.

**Comment by AnthonyTheKoala** [comment]
Dear rkern,
Thank you for that.
A couple of additional comments which I hope contributes back to the scipy people:
* A few weeks ago, I'm sorry that is vague, in idle one was able to _import scipy_ and  _dir(scipy)_ listing all functions or do a complete autocomplete until now.
* On the "DLL" error, to cut a long story short, that is after research I installed the MKL version of the numpy with mkl at https://www.lfd.uci.edu/~gohlke/pythonlibs/#numpy.
and for my Python v 3.9.12, installed _numpy‑1.22.3+mkl‑cp39‑cp39‑win_amd64.whl_
* After installing the above *.whl file I observed the warning notices and downgraded the recommended packages, for example _cython_
* Then _pip uninstall scipy_ and then _pip install scipy --force-reinstall_
* The "DLL" error disappeared.
* This is what we cannot and can do by instantiating directly and indirectly:
 
![image](https://user-images.githubusercontent.com/26536316/162098411-87f79b89-18b3-417c-af61-f4ef7131a706.png)

Thank you,
Anthony of Sydney
